bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,176 @@
+        ]);
+
+        $batchSize = 50;
+
+        $searchResults = $this->searchService->findContent($query);
+        $totalResults = $searchResults->totalCount ?? 0;
+
+        $this->style->newLine();
+        $this->style->progressStart($totalResults);
+
+        $fieldIdentifier = $input->getOption('field-identifier');
+
+        for ($offset = 0; $offset < $totalResults; $offset += $batchSize) {
+            $query->offset = $offset;
+            $query->limit = $batchSize;
+            $searchResults = $this->searchService->findContent($query);
+
+            $this->repository->beginTransaction();
+
+            foreach ($searchResults->searchHits as $searchHit) {
+                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                $content = $searchHit->valueObject;
+
+                if ($content->getField($fieldIdentifier) === null) {
+                    continue;
+                }
+
+                try {
+                    if (!$content->getField($fieldIdentifier)->value instanceof TagFieldValue) {
+                        $this->style->warning(sprintf('Field with identifier %s must be a type of eztags', $fieldIdentifier));
+
+                        continue;
+                    }
+
+                    $alreadyAssignedTags = $content->getFieldValue($fieldIdentifier)->tags;
+                    $tag = $this->tagsService->loadTag($this->getTagId((int) $input->getOption('tag-id')));
+                    $tagsToAssign = array_filter($alreadyAssignedTags, static fn ($alreadyAssignedTag) => $tag->id !== $alreadyAssignedTag->id);
+                    $tagsToAssign[] = $tag;
+
+                    $this->repository->sudo(
+                        function () use ($content, $fieldIdentifier, $tagsToAssign): void {
+                            $contentDraft = $this->contentService->createContentDraft($content->contentInfo);
+                            $contentUpdateStruct = $this->contentService->newContentUpdateStruct();
+
+                            $contentUpdateStruct->setField($fieldIdentifier, new TagFieldValue($tagsToAssign));
+
+                            $this->contentService->updateContent($contentDraft->versionInfo, $contentUpdateStruct);
+                            $this->contentService->publishVersion($contentDraft->versionInfo);
+                        },
+                    );
+
+                    $this->style->progressAdvance();
+                } catch (Throwable $t) {
+                    $this->style->error($t->getMessage());
+
+                    $this->repository->rollback();
+
+                    continue;
+                }
+            }
+            $this->repository->commit();
+        }
+
+        $this->style->progressFinish();
+        $this->style->success('Tags assigned successfully');
+
+        return Command::SUCCESS;
+    }
+
+    private function getParentLocation(int $parentLocationInput): int
```

Comment:
If we're using `(int) $input->getOption('tag-id'))` to directly load a tag, do we need this method? Why not just suply the casted option to the query and let Ibexa let us know of any errors?